docs/site/DEVELOPING.md

```diff
@@ -286,6 +286,46 @@ LoopBack 4 documentation is hosted inside this monorepo in the
 This allows us to change both implementation and the documentation in a single
 pull request.
 
+### Organization of content
+
+LoopBack 4 is a large codebase organized into dozens of packages. We,
+developers, are often tempted to include a lot of details and wider context when
+writing docs. While this is good for capturing the knowledge in a text form, it
+also makes the documentation difficult to follow by people new to LoopBack.
+Newcomers get quickly overwhelmed by the amount of concepts they need to
+understand and all the different packages they need to know about.
+
+To prevent that issue, we have categorized all monorepo packages into two
+groups:
+
+1. Building-blocks are lower-level packages that are typically not consumed by
+   LoopBack projects directly, instead there is a higher-level package exposing
+   the functionality and/or the public API of these building blocks.
+
+2. Framework-level packages are used directly by LoopBack projects and provide
+   API that LoopBack consumers use.
+
+In our documentation, CLI template and example applications, you should always
+refer to framework-level packages. Building-block packages should have their own
+documentation, for example in their `README.md` or in a dedicated section of
+[loopback.io](https://loopback.io).
+
+The rule of thumb: if a higher-level package is re-exporting all public APIs
+from a lower-level package (e.g. `@loopback/core` re-exports
+`@loopback/context`) or is using another package as a implementation building
+block (e.g. `@loopback/rest` uses `@loopback/express`), then the lower-level
+package should be moved to "building blocks" and our documentation should be
+updated to reference the higher-level "framework" instead.
+
+List of packages that are considered as building blocks:
```

Comment:
And the rest are framework-level pacakges? If yes, we should mention that, else it feels like the list of framework-level pacakges is missing.

Reply:
Done in 69dcaa6

There is one thing that bugs me a bit: we also have utility packages like `@loopback/build` or `@loopback/testlab` that are technically not part of the framework, but they are not building blocks either in the sense that they should be mentioned in our framework documentation.

I guess "framework-level" is not the best term to use. Can we come up with a more accurate name?

Reply:
What about the `extensions` and `examples` packages? Those seem more like an adds-on.

Reply:
> What about the `extensions` and `examples` packages? Those seem more like an adds-on.

Sure :+1:

What I am trying to accomplish here is to distinguish between "building blocks" packages that should not be referred to in our docs, CLI templates and example apps; and everything else. 

I used the term "framework-level" for "everything else", which is clearly misleading based on your feedback.

So, what would be a better term?

Reply:
> I guess "framework-level" is not the best term to use. Can we come up with a more accurate name?

They need not belong in the framework packages. They can be classified as helper/utility packages.

Reply:
I am adding the following clarification to the docs:

```
#### Framework-level packages

All packages not listed above are considered as framework-level.

We consider utilities like `@loopback/testlab` and example projects like
`@loopback/todo` as framework-level too.```

Reply:
> We consider utilities like `@loopback/testlab` and example projects like
`@loopback/todo` as framework-level too.

Doesn't feel right. Anything that LoopBack developers would not (maybe even never) use in their project feels very odd to be classified under framework-level packages. We need a third category - helpers, utility, along that line.

Reply:
I guess this is subjective.

Let's land my PR as the first step, and then you can open a follow-up PR to discuss how to improve this initial categorization.

Reply:
Sounds good.